tools/MakeNiiFilesBIDSCompliant.pl

```diff
@@ -0,0 +1,705 @@
+            my ($extraHeader, $extraHeaderVal);
+
+            foreach my $j (0..scalar(@headerNameMINCArr)-1) {
+                $headerNameMINC = $headerNameMINCArr[$j];
+                $headerName   = $headerNameArr[$j];
+                $headerName   =~ s/^\"+|\"$//g;
+                print "Adding now $headerName header to $headerFile\n" if $verbose;
+
+                $headerVal    =   &fetchMincHeader($mincFileName,$headerNameMINC);
+                # Some headers need to be explicitely converted to floats in Perl
+                # so json_encode does not add the double quotation around them
+                if (($headerNameMINC eq 'acquisition:repetition_time') ||
+                    ($headerNameMINC eq 'acquisition:echo_time') ||
+                    ($headerNameMINC eq 'acquisition:inversion_time') ||
+                    ($headerNameMINC eq 'dicom_0x0018:el_0x1314')) {
+                        $headerVal *= 1;
+                }
+
+
+#                if (defined($headerVal) && ($headerVal ne '')) {
+#                if (defined($headerVal) && ($headerVal != '')) {
+                if (defined($headerVal)) {
+                    $header_hash{$headerName} = $headerVal;
+                        print "     $headerName was found for $mincFileName with value $headerVal\n" if $verbose;
+
+                    # If scanner is Philips, store this as condition 1 being met
+                    if ($headerNameMINC eq 'study:manufacturer' && $headerVal =~ /Philips/i) {
+                        $manufacturerPhilips = 1;
+                    }
+                }
+                else {
+                    print "     $headerName was not found for $mincFileName\n" if $verbose;
+                }
+            }
+            
+            # If manufacturer is Philips, then add SliceOrder to the JSON manually
+            ######## This is just for the BETA version #########
+            ## See the TODO section for improvements needed in the future on SliceOrder ##
+            if ($manufacturerPhilips == 1) {
+                $extraHeader = "SliceOrder";
+                $extraHeader =~ s/^\"+|\"$//g;
+                if ($sliceOrderPhilips) {
+                    $extraHeaderVal = $sliceOrderPhilips;
+                }
+                else {
+                    print "   This is a Philips Scanner with no $extraHeader
+                    defined at the command line argument 'slice_order_philips'.
+                    Logging in the JSON as 'Not Supplied' \n" if $verbose;
+                }
+                $header_hash{$extraHeader} = $extraHeaderVal;
+                    print "    $extraHeaderVal was added for Philips Scanners'
+                    $extraHeader \n" if $verbose;
+            } else {
+                # get the SliceTiming from the proper header
+                # split on the ',', remove trailing '.' if exists, and add [] to make it a list
+                $headerNameMINC = 'dicom_0x0019:el_0x1029';
```

Comment:
are you sure this is the right header?

For my DWI file, this header stores the following:
```
		dicom_0x0019:el_0x1029 = 0b, 0b, 0b, 0b, 0b, 0b, 0b, 0b, -26b, 85b, 0b, 0b, 0b, -124b, -78b, 64b, -57b, -68b, 10b, 0b, 0b, -48b, 97b, 64b, -13b, 42b, 0b, 0b, -128b, 18b, -77b, 64b, 50b, -81b, 2b, 0b, 0b, -8b, 113b, 64b, 0b, 0b, 0b, 0b, -128b, -93b, -77b, 64b, 100b, 94b, 5b, 0b, 0b, -32b, 122b, 64b, -13b, 42b, 0b, 0b, 0b, 50b, -76b, 64b, -103b, 87b, 1b, 0b, 0b, -8b, -127b, 64b, 0b, 0b, 0b, 0b, 0b, -61b, -76b, 64b, 0b, 0b, 0b, 0b, 0b, 108b, -122b, 64b, -26b, 85b, 0b, 0b, -128b, 81b, -75b, 64b, 50b, -81b, 2b, 0b, 0b, -32b, -118b, 64b, -13b, 42b, 0b, 0b, -128b, -30b, -75b, 64b, 0b, 0b, 0b, 0b, 0b, 104b, -113b, 64b, 0b, 0b, 0b, 0b, 0b, 113b, -74b, 64b, -103b, 87b, 1b, 0b, 0b, -18b, -111b, 64b, -13b, 42b, 0b, 0b, -128b, -1b, -74b, 64b, -52b, -85b, 0b, 0b, 0b, 50b, -108b, 64b, 0b, 0b, 0b, 0b, -128b, -112b, -73b, 64b, 0b, 0b, 0b, 0b, 0b, 108b, -106b, 64b, -26b, 85b, 0b, 0b, 0b, 31b, -72b, 64b, -103b, 87b, 1b, 0b, 0b, -80b, -104b, 64b, -13b, 42b, 0b, 0b, 0b, -80b, -72b, 64b, 0b, 0b, 0b, 0b, 0b, -22b, -102b, 64b, -26b, 85b, 0b, 0b, -128b, 62b, -71b, 64b, -103b, 87b, 1b, 0b, 0b, 46b, -99b, 64b, -13b, 42b, 0b, 0b, -128b, -49b, -71b, 64b, -52b, -85b, 0b, 0b, 0b, 104b, -97b, 64b, 0b, 0b, 0b, 0b, 0b, 94b, -70b, 64b, 0b, 0b, 0b, 0b, 0b, -42b, -96b, 64b, -26b, 85b, 0b, 0b, 0b, -17b, -70b, 64b, -52b, -85b, 0b, 0b, 0b, -13b, -95b, 64b, -13b, 42b, 0b, 0b, -128b, 125b, -69b, 64b, 0b, 0b, 0b, 0b, 0b, 16b, -93b, 64b, 0b, 0b, 0b, 0b, -128b, 14b, -68b, 64b, -52b, -85b, 0b, 0b, 0b, 50b, -92b, 64b, -13b, 42b, 0b, 0b, 0b, -99b, -68b, 64b, -26b, 85b, 0b, 0b, 0b, 79b, -91b, 64b, 0b, 0b, 0b, 0b, -128b, 43b, -67b, 64b, 0b, 0b, 0b, 0b, 0b, 113b, -90b, 64b, -26b, 85b, 0b, 0b, -128b, -68b, -67b, 64b, -26b, 85b, 0b, 0b, 0b, -114b, -89b, 64b, 0b, 0b, 0b, 0b, 0b, 75b, -66b, 64b, 0b, 0b, 0b, 0b, 0b, -80b, -88b, 64b, -26b, 85b, 0b, 0b, 0b, -36b, -66b, 64b, -52b, -85b, 0b, 0b, 0b, -51b, -87b, 64b, -13b, 42b, 0b, 0b, -128b, 106b, -65b, 64b, -26b, 85b, 0b, 0b, 0b, -17b, -86b, 64b, 0b, 0b, 0b, 0b, -128b, -5b, -65b, 64b, 0b, 0b, 0b, 0b, 0b, 12b, -84b, 64b, -13b, 42b, 0b, 0b, 0b, 69b, -64b, 64b, -52b, -85b, 0b, 0b, 0b, 46b, -83b, 64b, 122b, 21b, 0b, 0b, -128b, -115b, -64b, 64b, 0b, 0b, 0b, 0b, 0b, 75b, -82b, 64b, -13b, 42b, 0b, 0b, -64b, -44b, -64b, 64b, -52b, -85b, 0b, 0b, 0b, 104b, -81b, 64b, 122b, 21b, 0b, 0b, 64b, 29b, -63b, 64b, -13b, 42b, 0b, 0b, 0b, 69b, -80b, 64b, 0b, 0b, 0b, 0b, -128b, 100b, -63b, 64b, 0b, 0b, 0b, 0b, -128b, -45b, -80b, 64b, -13b, 42b, 0b, 0b, 0b, -83b, -63b, 64b, -13b, 42b, 0b, 0b, -128b, 100b, -79b, 64b, 0b, 0b, 0b, 0b, 64b, -12b, -63b, 64b, 0b, 0b, 0b, 0b, 0b, -13b, -79b, 64b ;
```
because of that, the line 542 below is failing for my DWI file as it cannot convert a string into a float.

Reply:
mmm. Same storing for my BOLD file...

Reply:
do you have access to the DICOMs?
I am pretty sure I saw this before and it was a dcm2mnc issue as the DICOMs were stored properly...
but since I used a new dcm2mnc converter never saw this problem again...

Reply:
ok. Just checked on all MINC files I had and the results are all in the same format for that header... Not sure why.